x-pack/plugins/viz_editor/essql_datasource_plugin/index_patterns.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import chrome from 'ui/chrome';
+import { Datasource } from '../public/common/lib';
+
+export const getIndexPatterns = () => {
+  const savedObjectsClient = chrome.getSavedObjectsClient();
+  return savedObjectsClient
+    .find({
+      type: 'index-pattern',
+      perPage: 1000, // TODO: Paginate index patterns
+    })
+    .then(resp => {
+      return resp.savedObjects.map(({ id, attributes }) => {
+        return Object.assign(attributes, {
+          id,
+          title: attributes.title,
+          fields: JSON.parse(attributes.fields as string),
+        }) as Datasource;
+      });
+    })
+    .catch(err => {
+      // TODO: Show errors to users
+    });
```

Comment:
I'm a bit unsure why you need this for the SQL plugin?

Reply:
Got copy-pasted